src/main/java/uk/ac/cam/cl/dtg/segue/api/QuestionFacade.java

```diff
@@ -232,6 +261,69 @@ public Response getQuestionsAnswered(@Context final HttpServletRequest request,
     }
   }
 
+  /**
+   * REST end point to provide five questions in random located in the question tile of the student dashboard.
+   *
+   * @param request  - this allows us to check to see if a user is currently logged in.
+   * @param subjects - a comma separated list of subjects
+   * @return a Response containing a gameboard object or containing a SegueErrorResponse.
+   */
+  @GET
+  @Path("/randomQuestions")
+  @Produces(MediaType.APPLICATION_JSON)
+  @GZIP
+  public final Response getRandomQuestions(@Context final HttpServletRequest request,
+                                           @QueryParam("subjects") final String subjects) {
+    try {
+      RegisteredUserDTO currentUser = this.userManager.getCurrentRegisteredUser(request);
+      var filterQuestionsPreference = this.userPreferenceManager.getUserPreference(
+          "DISPLAY_SETTING",
+          "HIDE_NON_AUDIENCE_CONTENT",
+          currentUser.getId());
+
+      GameFilter gameFilter = new GameFilter();
+
+      if (filterQuestionsPreference != null && filterQuestionsPreference.getPreferenceValue()) {
+        var userContexts = currentUser.getRegisteredContexts();
+
+        List<String> subjectsList = splitCsvStringQueryParam(subjects);
+        List<String> stagesList = new ArrayList<>();
+        List<String> examBoardsList = new ArrayList<>();
+
+        for (UserContext uc : userContexts) {
+          stagesList.add(uc.getStage().name());
+          examBoardsList.add(uc.getExamBoard().name());
+        }
+
+        gameFilter = new GameFilter(
+            subjectsList,
+            null,
+            null,
+            null,
+            null,
+            null,
+            stagesList,
+            null,
+            examBoardsList);
+      }
+
+      var questions = this.gameManager.generateRandomQuestions(gameFilter, 5);
+
+      // Return the list of random questions as JSON
+      return Response.ok(questions).build();
+    } catch (NoUserLoggedInException e) {
+      return SegueErrorResponse.getNotLoggedInResponse();
+    } catch (ContentManagerException e) {
+      return new SegueErrorResponse(Status.NOT_FOUND, "Error creating random questions")
```

Comment:
There is a mismatch between the status code (404) and the message: why are we returning a 404 when this looks like an internal error?

Reply:
Agreed. This was requested by the FE Dev. What do you feel about "Questions not found" as this error will only show if there are less than 5 questions generated based on a filter?

Reply:
The actual message I have no preference on wording, the only request was to return "something like a 404 error". Any error status code under 500 will display a popup error on the front end and default to an alternative component which is a far better user experience. If it returns 500 it takes out the whole page with a "We're sorry, but an error has occurred on the Isaac server!"

Reply:
Thanks for your comments, I think I understand this better now. It's a bit tricky, because it sounds like there are only two valid states: we want to either return 5 questions, or nothing at all.

For the use case in which there are less than 5 questions, we cannot return a 4xx error, because that's reserved for client errors, and this is not the case. So I would suggest to either:
- Return a 204 (No Content).
- Or even better, just return as many questions as you can (up to 5). Then let the frontend decide what to do. If you're returning an empty array, or an array with, say, 3 questions, the frontend can check the length and act accordingly.

Reply:
Front end now has logic for handling different lengths of content returned, including defaulting to an alternative component if the response is "successful" but the data is an empty array.  This theoretically means that if the API was changed to return 10 questions in the future for example, this would work without any front end changes needed.